caddy/config_example.json

```diff
@@ -10,7 +10,42 @@
     "apps": {
         "http": {
             "servers": {
-				"": {
+                "srv0": {
+                    "listen": [":8000"],
+                    "routes": [
+                        {
+                            "match": [
+								{
+									"path": [
+										"/tcp"
+									]
+								}
+							],
+                            "handle": [
+                                {
+                                    "handler": "websocket",
+                                    "backend": ":9000"
```

Comment:
Note that this drops the client IP. We would need something like the PROXY protocol, in which case we should use unix sockets instead of TCP/UDP. But there's a better way.

We should create a a websocket Layer-4 reverse proxy instead, that plugs directly into the layer-4 handler and avoid extra dialing and relaying. It also remove the need for a port.

ws2endpoint can serve as inspiration: https://github.com/Jigsaw-Code/outline-sdk/blob/fe5dc64d1503712f68d53085aa5e53e3d0ea7270/x/examples/ws2endpoint/main.go#L71

Though I see it as being a connection that wraps the http.ResponseWriter and http.Request and is passed to the SS handler.

Reply:
You're right; moved it to Layer-4.

I still need to get UDP to work, that's been causing me some trouble, but in the meantime would love to get your thoughts on this draft.